### PR TITLE
Allow .yaml and .yml extension for azure-dev pipeline files

### LINKED ISSUE #4170: [Issue] azd pipeline config should also check if azure-dev.yaml
- [X] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install)

**Output from `azd version`**
Run `azd version` and copy and paste the output here:

**Describe the bug**

When I run azd pipeline config in https://github.com/Azure-Samples/openai-chat-vision-quickstart/, I see this:

```
openai-chat-vision-quickstart % azd pipeline config

? Select a provider: GitHub

The default azure-dev.yml file, which contains a basic workflow to help you get started, is missing from your project.
```

My project *does* have an azure-dev.yaml however. I name all my YAMLS *.yaml after conferring with the community around the proper extension. My community replies may have been wrong, but that's what I've been doing since.

Anyway, it'd be great if it could also look for yaml. Alternatively, I need to rename every workflow file so that people don't get confused in my template repos.

### PR DESCRIPTION
**Pull request by jongio**
Fixes #4170

### CODE REVIEW

**Comment by jongio** [comment]
@pamelafox - Can you test this build for your scenario?

File: cli/azd/pkg/pipeline/pipeline_manager.go [hunk 1]
```diff
@@ -998,12 +971,23 @@ func generatePipelineDefinition(path string, props projectProperties) error {
 	return nil
 }
 
+// hasPipelineFile checks if any pipeline files exist for the given provider in the specified repository root.
+func hasPipelineFile(provider, repoRoot string) bool {
```
Comment (vhvb1989) [inline cli/azd/pkg/pipeline/pipeline_manager.go:975]: make provider to be `ciProviderType`

File: cli/azd/pkg/pipeline/pipeline.go [hunk 2]
```diff
@@ -160,26 +160,61 @@ func mergeProjectVariablesAndSecrets(
...
-	gitHubWorkflowsDirectory string = filepath.Join(gitHubDirectory, "workflows")
-	azdoPipelinesDirectory   string = filepath.Join(azdoDirectory, "pipelines")
-	gitHubYml                string = filepath.Join(gitHubWorkflowsDirectory, defaultPipelineFileName)
-	azdoYml                  string = filepath.Join(azdoPipelinesDirectory, defaultPipelineFileName)
+	pipelineFileNames = []string{"azure-dev.yml", "azure-dev.yaml"}
 )
 
+var (
+	// Define a map to hold the directory and file names for each provider
+	pipelineProviderFiles = map[string]struct {
```
Comment (vhvb1989) [inline cli/azd/pkg/pipeline/pipeline.go:180]: use `map[ciProviderType]struct`

**Comment by jongio** [comment]
> @pamelafox - Can you test this build for your scenario?

I tested this with azure-dev.yaml, azure-dev.yml and it works.

File: cli/azd/pkg/pipeline/pipeline_manager.go [hunk 3]
```diff
@@ -822,71 +822,55 @@ func (pm *PipelineManager) savePipelineProviderToEnv(
...
 }
 
+func validateProvider(provider ciProviderType) error {
+	if provider == "" {
+		errMsg := "Unknown CI provider"
+		log.Println("Error:", errMsg)
+		return fmt.Errorf(errMsg)
+	}
+	return nil
+}
```
Comment (vhvb1989) [inline cli/azd/pkg/pipeline/pipeline_manager.go:832]: You can remove this. 
Checking for a valid provider happens on the beginning of pipeline config.  This code will never happen.
It is safe to assume the provider will be always valid at this point

File: cli/azd/pkg/pipeline/pipeline_manager.go [hunk 4]
```diff
@@ -822,71 +822,55 @@ func (pm *PipelineManager) savePipelineProviderToEnv(
...
-		errMsg := fmt.Sprintf("%s is not a known pipeline provider", props.CiProvider)
-		log.Println("Error:", errMsg)
-		return fmt.Errorf(errMsg)
+	if err := validateProvider(props.CiProvider); err != nil {
+		return err
 	}
 
-	log.Printf("YAML path: %s", providerCheck.ymlPath)
-	log.Printf("Directory path: %s", providerCheck.dirPath)
+	provider := string(props.CiProvider)
```
Comment (vhvb1989) [inline cli/azd/pkg/pipeline/pipeline_manager.go:842]: do we really need to cast here?  If this is for calling `log.Printf()`, you can pass props.CiProvider directly

**Review by vhvb1989 — APPROVED** [review]
little suggestions, otherwise LGTM

File: cli/azd/pkg/pipeline/pipeline_manager.go (hunk 4 above)
Comment (jongio) [reply cli/azd/pkg/pipeline/pipeline_manager.go]: Removed

**Comment by pamelafox** [comment]
I ran this on my repo from before, it works as expected! Thank you :)

**Pull request merged**